### PR TITLE
Reset expanded directories list on category change

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Pull Request Description

Partially closes: cloud-v2/1592

Closes: https://github.com/enso-org/cloud-v2/issues/1606

This PR also adds needed configuration for unit tests and adjust it to run using vscode vite extension

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -568,4 +557,22 @@ export default [
       '@typescript-eslint/naming-convention': 'off',
     },
   },
+  {
+    files: ['app/gui/src/dashboard/**/*.test.tsx', 'app/gui/src/dashboard/**/*.test.ts'],
```
Comment (MrFlashAccount) [inline eslint.config.mjs:561]: Loosening rules for tests, as they mostly make no sense there

File: tsconfig.json [hunk 2]
```diff
@@ -21,5 +21,6 @@
     "skipLibCheck": true,
     "target": "ES2022",
     "jsx": "react-jsx"
-  }
+  },
+  "exclude": ["tools", "node_modules", "lib", "engine", "test", "dist", "docs"]
```
Comment (MrFlashAccount) [inline tsconfig.json:25]: Questionable, but advised by vscode typescript to add this.
Improves reference discovery time

File: app/gui/vite.config.ts [hunk 3]
```diff
@@ -57,11 +61,8 @@ export default defineConfig({
       include: fileURLToPath(new URL('./src/dashboard/**/*.tsx', import.meta.url)),
       babel: {
         plugins: [
-          '@babel/plugin-syntax-import-attributes',
-          [
-            'babel-plugin-react-compiler',
-            { target: '18', enablePreserveExistingMemoizationGuarantees: true },
-          ],
+          syntaxImportAttributes,
```
Comment (MrFlashAccount) [inline app/gui/vite.config.ts:64]: For some reason, vitest extention fails to resolve these reds when they specified as strings (guess it tries to search them in the root of the monorepo). Using explicit import solves it

**Review by PabloBuchu — APPROVED** [review]

File: app/gui/vite.config.ts (hunk 3 above)
Comment (somebody1234) [reply app/gui/vite.config.ts:64]: off topic (out of scope), but i wonder why we still need import attributes? i believe they are stable in node 22, i think?

**Review by somebody1234 — APPROVED** [review]
CR :white_check_mark:

File: app/gui/vite.config.ts (hunk 3 above)
Comment (MrFlashAccount) [reply app/gui/vite.config.ts:64]: I think no longer we need this, as it's included in the preset-env:https://babeljs.io/docs/babel-plugin-syntax-import-attributes

**Pull request merged**